src/octoprint/util/comm.py

```diff
@@ -4641,6 +4720,26 @@ def parse_resend_line(line):
 
 	return None
 
+def parse_advanced_ok_line(line):
+	"""
+	Parses the provided marlin ADVANCED_OK line.
+	Args:
+		line (str): the line to parse
+	Returns:
+    last_processed_line
```

Comment:
From the looks of it this should rather be

``` python
def parse_advanced_ok_line(line):
    """
    Parses the provided ADVANCED_OK line.

    Format:
        ok N<last processed line> P<free planner slots> B<free command slots>

    Args:
        line (str): the line to parse

    Returns:
        (dict or None) A dict containing the last processed line number (``N``), the amount of free 
        slots in the motion planner buffer (``P``) and the amount of free slots in the command buffer (``B``), 
        or None if no match could be made.
    """ 
    # ...
```

And something about the whitespace seems to be fishy in the function as well.